File: Java/NetBeans2/QLNV/src/qlnv/NVVP.java
package qlnv;

public class NVVP extends NV {

    private float heSoLuong;

    public NVVP(int maNV, String hoTen, int namSinh, int soNamCT, float luongCB, float heSoLuong) {
        super(maNV, hoTen, namSinh, soNamCT, luongCB);
        this.heSoLuong = heSoLuong;
    }

    public float tinhLuong() {
        float phuCap;

        if (soNamCT < 10) {
            phuCap = luongCB;
        } else {
            phuCap = 1.2f * luongCB;
        }

        return heSoLuong * luongCB + phuCap;
    }

    public String loaiNV() {
        return "VP";
    }

    public String toString() {
        return super.toString() + ";" + heSoLuong;
    }

    public boolean equals(Object obj) {
        if (obj instanceof NVVP) {
            NVVP nvvp = (NVVP) obj;

            if (super.equals(nvvp) && heSoLuong == nvvp.heSoLuong) {
                return true;
            }
        }

        return false;
    }
}

File: C#/giuaKY/giuaKY/NVVP.cs
namespace giuaKY
{
    public class NVVP : NV
    {
        private float heSoLuong;

        public NVVP(int maNV, string hoTen, int namSinh, int soNamCT, float luongCB, float heSoLuong) : base(maNV, hoTen, namSinh, soNamCT, luongCB)
        {
            this.heSoLuong = heSoLuong;
        }

        public override void nhap()
        {
            base.nhap();
            Console.Write("Nhap he so luong: ");
            heSoLuong = Convert.ToSingle(heSoLuong);
        }

        public override string loaiNV()
        {
            return "NVVP";
        }

        public override float tinhLuong()
        {
            float phuCap;

            if (soNamCT < 10)
            {
                phuCap = luongCB;
            }
            else
            {
                phuCap = 1.2f * luongCB;
            }

            return heSoLuong * luongCB + phuCap;
        }

        public override string ToString()
        {
            return base.ToString() + ";" + heSoLuong;
        }
    }
}
